packages/cloud_firestore/lib/src/blob.dart

```diff
@@ -0,0 +1,17 @@
+// Copyright 2018, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of cloud_firestore;
+
+class Blob {
+  final Uint8List bytes;
+  const Blob(this.bytes);
+
+  @override
+  bool operator ==(dynamic other) =>
+      other is Blob && DeepCollectionEquality().equals(other.bytes, bytes);
+
+  @override
+  int get hashCode => bytes.hashCode;
```

Comment:
That `hashCode` is probably  based on the reference identity of `bytes`, not its contents. Which means it will be inconsistent with your `operator==`. You can use `hashValues` from `dart:ui`.